### PR TITLE
Serde for Vec[2, 3, 4] and Mat[2, 3, 4]

### PR DESCRIPTION
**Pull request by MagicRB**
Haven't really tested it, but it does compile and it shouldn't have any issues, if they exist I'm going to find them soon, since I'll be using this in my project.

### CODE REVIEW

**Comment by MagicRB** [comment]
One more thing, I wasn't sure how to name the columns in Mat[2, 3, 4], so I just named them a, b, c, d

**Comment by aloucks** [comment]
> One more thing, I wasn't sure how to name the columns in Mat[2, 3, 4], so I just named them a, b, c, d

@MagicRB Why not name the wide matrix fields (and serialize them) the same way as the non-wide versions? In other words, the serialized versions would/should look identical (i.e. a `cols` field that is a sequence of vecs)

Alternatively, you could serialize everything as sequences (e.g. Vec3 `[f32; 3]` and Mat4 `[f32; 16]`)

@termhn What do you think?

**Comment by termhn** [comment]
I think it makes sense to serialize/deserialize matrices as containing simply `[f32; 16]` or whatever the equivalent number of floats is, and same for Vecs.

**Comment by MagicRB** [comment]
OK, it should now be ready to merge, now Mat[2, 3, 4] are serialized as simple f32 arrays, Vec[2,3,4] still as named structs. If there are any issue with my git usage, tell me and I'll fix them, I'm still new to all this git stuff.

**Comment by MagicRB** [comment]
I may get around to adding support for wide types and getting in place deserialization (or serialization can't remember which one) to work. No ETA given, so you might want to wait before merging this, also I'm working on implementing PartiaEq which will help with tests

File: Cargo.toml [hunk 1]
```diff
@@ -11,4 +11,9 @@ license = "MIT OR Apache-2.0 OR Zlib"
 
 [dependencies]
 # wide = { path = "../wide" }
-wide = "0.4"
\ No newline at end of file
+wide = "0.4"
+serde = { version = "1.0", features = [], optional = true }
+serde_test = { version = "1.0", features = [], optional = true }
```
Comment (aloucks) [inline Cargo.toml]: It would be better to add `serde_test` as a `dev-dependency` instead and remove it here.

File: Cargo.toml [hunk 2]
```diff
@@ -11,4 +11,9 @@ license = "MIT OR Apache-2.0 OR Zlib"
...
+
+[features]
+serde-support = ["serde", "serde_test"]
```
Comment (aloucks) [inline Cargo.toml]: You can remove this as a feature and the feature will just be `serde`.

**Comment by MagicRB** [comment]
Will do

**Comment by MagicRB** [comment]
Ok, I fixed everything you pointed out and force pushed it. If we merge PartialEq before this, I can actually write proper tests

**Comment by termhn** [comment]
Closed in favor of #38, thanks for doing the original work on this @MagicRB !

**Pull request closed without merging**